### PR TITLE
refactor(comp:list): rebuild with tsx (#187)

### PR DESCRIPTION
**Pull request by qingshan-dev**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/IDuxFE/idux/blob/main/packages/site/src/docs/Contributing.zh.md#commit)
- [x] Tests for the changes have been added/updated or not needed
- [x] Docs and demo have been added/updated or not needed

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [x] Component style update
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

### CODE REVIEW

**Review by qingshan-dev — COMMENTED** [review]
Self check is finish

File: packages/components/list/src/List.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      }
+      emit('loadMore', done)
+    }
+
+    const useClasses = (props: ListProps, listConfig: ListConfig) => {
+      return computed(() => {
+        const borderless = props.borderless ?? listConfig.borderless
+        const size = props.size ?? listConfig.size
+        return [
+          'ix-list',
```
Comment (danranVm) [inline packages/components/list/src/List.tsx:59]: `ix` 的前缀别写死，参考一下其他组件的。

File: packages/components/list/src/List.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      return props.loading
+    })
+    const loadMoreLoading = ref(false)
+    const loadMoreHandleClick = () => {
+      loadMoreLoading.value = true
+      const done = () => {
+        loadMoreLoading.value = false
+      }
+      emit('loadMore', done)
+    }
```
Comment (danranVm) [inline packages/components/list/src/List.tsx:52]: 这些都不要吧，内部不去维护这个状态了，用户自己控制 loading 就好了。

**Review by danranVm — CHANGES_REQUESTED** [review]

File: packages/components/list/src/List.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+export default defineComponent({
+  name: 'IxList',
+  components: {
+    IxSpin,
+    IxButton,
+    IxEmpty,
+  },
+  props: listProps,
+  emits: ['loadMore'],
```
Comment (danranVm) [inline packages/components/list/src/List.tsx:28]: 这个干掉，如果有事件的话，声明在 props 中，参考一下其他组件。

File: packages/components/list/src/List.tsx (hunk 1 above)
Comment (qingshan-dev) [reply packages/components/list/src/List.tsx]: 已使用common.prefixCls

File: packages/components/list/src/List.tsx (hunk 2 above)
Comment (qingshan-dev) [reply packages/components/list/src/List.tsx]: 使用 footer 替换使用, 减少内部代码体积

File: packages/components/list/src/List.tsx (hunk 3 above)
Comment (qingshan-dev) [reply packages/components/list/src/List.tsx]: 已删除

**Review by danranVm — APPROVED** [review]
LGTM

**Pull request merged**